src/main/scala/thepackage/services/UserService.scala

```diff
@@ -0,0 +1,89 @@
+package thepackage.services
+
+import java.time.Clock
+import java.util.UUID
+
+import cats.data.NonEmptyChain
+import cats.data.Validated.{Invalid, Valid}
+import thepackage.models.User
+import thepackage.payloads.requests.CreateUserRequest
+import thepackage.util.Validatable
+import cats.data._
+import cats.syntax._
+import cats.implicits._
+import thepackage.errors._
+import thepackage.util._
+import thepackage.db.PostgresProfile.api._
+import thepackage.db.PostgresProfile.backend.Database
+import thepackage.db.tables.UserTable
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait UserService {
+  def createUser(request: CreateUserRequest): Future[Either[UserCreationError, User]]
+  def retrieveUser(id: UUID): Future[Option[User]]
+  def retrieveUserByCredentials(email: String, password: String): Future[Option[User]]
+  def updateUserPassword(userId: UUID, newPassword: String): Future[Any]
+}
+
+class DefaultUserService(db: Database, passwordService: PasswordService[String])
+  (implicit ec: ExecutionContext, clock: Clock)
+  extends UserService {
+
+  implicit object CreateUserValidatable extends Validatable[CreateUserRequest, RequestValidationError, User] {
+    def validateEmail(email: String) =
+      email.split('@') match {
+        case parts if parts.length != 2 => FieldValidationError("email", "Exactly one @ symbol allowed").invalidNec
+        case Array(_, domain, _*) if !domain.contains('.') =>
```

Comment:
TLD check? blacklist check?

Reply:
I've found that none of that has a meaningful impact. The only way to go is a verification email in the on-boarding flow which I'm figuring users of this will implement later.

But maybe I should add a generic email client trait with a default no-op implementation for sending out verification emails. Then users can just implement the email client for whatever transactional mail vendor they end up with.
Thoughts?